### PR TITLE
Feature/error handling eco news

### PR DESCRIPTION
**Pull request by HryshkoAnton**
Show notification with error message if there are any errors in Eco News Component

https://github.com/ita-social-projects/GreenCity/issues/1288

### CODE REVIEW

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 1]
```diff
@@ -8,6 +9,7 @@ import * as fromApp from '@store/app.reducers';
 import * as fromEcoNews from '@eco-news-store/eco-news.actions';
 import { EcoNewsSelectors } from '@eco-news-store/eco-news.selectors';
 import { EcoNewsDto } from '@eco-news-models/eco-news-dto';
+import { MatSnackBarComponent } from './../../../errors/mat-snack-bar/mat-snack-bar.component';
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:12]: Please, sort imports in such order: angular files, core files/libraries (RxJS etc.), local files. Speaking about this row, please, replace long dotted path with alias

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 2]
```diff
@@ -72,9 +75,21 @@ export class NewsListComponent implements OnInit, OnDestroy {
   private addElemsToCurrentList(): void {
     if (this.tagsList) {
       this.ecoNewsSubscription = this.ecoNewsService.getNewsListByTags(this.currentPage, this.numberOfNews, this.tagsList)
+        .pipe(
+          catchError((error) => {
+            this.snackBar.openSnackBar('Oops, something went wrong. Please reload page or try again later.', 'X', 'red-snackbar');
+            return error;
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:81]: Please, leave an ampty line before the 'return'

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 3]
```diff
@@ -72,9 +75,21 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
+          })
+        )
         .subscribe((list: EcoNewsDto) => this.setList(list));
     } else {
       this.ecoNewsSubscription = this.ecoNewsService.getEcoNewsListByPage(this.currentPage, this.numberOfNews)
+        .pipe(
+          catchError((error) => {
+            this.snackBar.openSnackBar('Oops, something went wrong. Please reload page or try again later.', 'X', 'red-snackbar');
+            return error;
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:90]: Same as line 81

File: src/app/component/eco-news/eco-news.module.ts [hunk 4]
```diff
@@ -25,6 +26,8 @@ import { EffectsModule } from '@ngrx/effects';
 import { EcoNewsEffects } from './store/eco-news.effects';
 import { EcoNewsSelectors } from './store/eco-news.selectors';
 import { CommentsModule } from '../comments/comments.module';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/eco-news.module.ts:29]: Please, sort imports in such order: angular files, core files/libraries (RxJS etc.), local files

Comment (HryshkoAnton) [reply src/app/component/eco-news/eco-news.module.ts]: fixed

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 3 above)
Comment (HryshkoAnton) [reply src/app/component/eco-news/components/news-list/news-list.component.ts:92]: fixed

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 2 above)
Comment (HryshkoAnton) [reply src/app/component/eco-news/components/news-list/news-list.component.ts:82]: fixed

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 1 above)
Comment (HryshkoAnton) [reply src/app/component/eco-news/components/news-list/news-list.component.ts]: fixed

File: src/app/component/errors/mat-snack-bar/mat-snack-bar.component.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  selector: 'app-mat-snack-bar',
+  templateUrl: './mat-snack-bar.component.html',
+  styleUrls: ['./mat-snack-bar.component.scss']
+})
+export class MatSnackBarComponent implements OnInit {
+
+  constructor(public snackBar: MatSnackBar) { }
+
+  ngOnInit() {
+  }
```
Comment (mplesha) [inline src/app/component/errors/mat-snack-bar/mat-snack-bar.component.ts:14]: empty method - in fact, you don't use this hook, so why do you put it here as well as in `import` and `implements`?

**Comment by mplesha** [comment]
conflicts in `tsconfig.json`

File: src/app/component/errors/mat-snack-bar/mat-snack-bar.component.ts (hunk 5 above)
Comment (HryshkoAnton) [reply src/app/component/errors/mat-snack-bar/mat-snack-bar.component.ts]: fixed

**Pull request merged**